pkg/scheduler/internal/cache/cache.go

```diff
@@ -462,6 +445,51 @@ func (cache *schedulerCache) removePod(pod *v1.Pod) error {
 	return nil
 }
 
+// Assumes that lock is already acquired.
+func (cache *schedulerCache) addPod(pod *v1.Pod, key string, isAssumed bool) {
+	cache.addPodToNode(pod)
+	ps := &podState{
+		pod: pod,
+	}
+	cache.podStates[key] = ps
+	if isAssumed {
+		cache.assumedPods.Insert(key)
+	}
+}
+
+// Assumes that lock is already acquired.
+func (cache *schedulerCache) updatePod(oldPod, newPod *v1.Pod, key string) error {
+	if err := cache.removePod(oldPod, key); err != nil {
+		return err
+	}
+
+	// A Pod delete event followed by an immediate Pod add event may be merged
+	// into a Pod update event. In this case, we should invalidate the old Pod, and
+	// then add the new Pod.
+	if oldPod.UID != newPod.UID {
+		// When the UID is different, we add the new Pod with the `newKey`
+		newKey, err := framework.GetPodKey(newPod)
+		if err != nil {
+			return err
+		}
+		key = newKey
+	}
+	cache.addPod(newPod, key, false)
+	return nil
+}
+
+// Assumes that lock is already acquired.
+func (cache *schedulerCache) removePod(pod *v1.Pod, key string) error {
+	if err := cache.removePodFromNode(pod); err != nil {
```

Comment:
`removePodFromNode()` is only used by this function, right? If so, flatten the logic reads better IMO. (you can add a comment btw)

Reply:
Yes, `removePodFromNode` is only used by `removePod` for now. But I’d like to keep this function for consistency and it may be good for more detailed unit test.

Reply:
we discourage unit test of low level functions, specially if they are only used by one other function.

I'm in favor of flattening too.

Reply:
flatten done